src/cron.ts

```diff
@@ -20,61 +19,120 @@ CronJob.from({
-        return false;
-      }
+  // Delete all old dateWritten from cache
+  cacheDb
+    .prepare(
+      `DELETE FROM date_written WHERE dateWritten < DATETIME('now', '-${range}');`
+    )
+    .run();
 
-      const startTime = new Date(startTimeStr);
+  // Write scored posts to cache
+  const writePosts = cacheDb.prepare(
+    `
+      INSERT INTO post (
+        did,
+        rkey,
+        url,
+        createdAt,
+        score,
+        raw_score,
+        decay,
+        likes,
+        reposts,
+        comments,
+        dateWritten
+      )
+      SELECT
+        did,
+        rkey,
+        url,
+        createdAt,
+        max(raw_score * decay) as score,
+        raw_score,
+        decay,
+        likes,
+        reposts,
+        comments,
+        DATETIME('now') as dateWritten
+      FROM
+        (
+          SELECT
+            local.post.did,
+            local.post.rkey,
+            local.post.url,
+            local.post.createdAt,
+            (1 - (unixepoch('now') - unixepoch(local.post.createdAt)) * 1.0 / 1000 / 60 / 60 / 24) as decay,
+            COUNT(local.reaction.type) AS raw_score,
+            COUNT(
+              CASE
+                WHEN local.reaction.type = 'like' THEN 1
+              END
+            ) AS likes,
+            COUNT(
+              CASE
+                WHEN local.reaction.type = 'repost' THEN 1
+              END
+            ) AS reposts,
+            COUNT(
+              CASE
+                WHEN local.reaction.type = 'comment' THEN 1
+              END
+            ) AS comments
+          FROM
+            local.post
+          LEFT JOIN local.reaction ON local.post.rkey = local.reaction.rkey
+          AND local.post.did = local.reaction.did
+        WHERE
+          local.post.createdAt >= DATETIME('now', '-${range}')
+        GROUP BY
+          local.post.did,
+          local.post.rkey,
+          local.post.url
+      )
+      GROUP BY
+        url
+      ORDER BY
+        score DESC;
+      `
+  );
 
-      return startTime.getTime() <= deleteCutoff.getTime();
-    });
+  const lastRun = writePosts.run();
+  cacheDb
+    .prepare(
+      `INSERT INTO date_written (dateWritten)
+      SELECT dateWritten FROM post WHERE rowid = ?;
+      `
+    )
+    .run(lastRun.lastInsertRowid);
 
-    const statement = db.prepare(
-      `DELETE FROM cache WHERE key in (${keysToDelete
-        .map((k) => `'${k}'`)
-        .join(", ")})`
-    );
+  console.log("WRITE POSTS", lastRun);
+};
 
-    console.log("DELETE CACHE", statement.run());
-  },
-});
+// write cache immediately at startup
+writeCache();
 
-// Prime the cache
+// Generate a new feed every 10 minutes
 CronJob.from({
   start: true,
-  // at 10:00:01 prime the cache
-  cronTime: "1 0/10 * * * *",
-  onTick: async () => {
-    console.log("Priming the cache....");
-    await Promise.all([
-      trendingLinks({ limit: 30 }),
-      trendingLinksHourly({ limit: 30 }),
-    ]);
-  },
+  cronTime: "0/10 * * * *",
```

Comment:
I also upped this to every 10 minutes instead of every hour

Reply:
Make sure you have a large enough disk on your server. Just from running it a few days off and on for testing, with less than 24 cache view writes my `cache.db` is like is 1.6GB. You know if we cut off the low score links, hell maybe only the 0 score links, that would likely cut off a ton of items from the cache. I'll test that out.